ballerina/tests/01-connection-init-test.bal

```diff
@@ -108,18 +110,144 @@ function testWithOptionsWithErroneousSSL() returns error? {
                     password: "password"
                 }
             }
      };
-    Client oracledbClient = check new(
+    Client|error oracledbClient = new(
         host = HOST,
         user = USER,
         password = PASSWORD,
         port = PORT,
         database = DATABASE,
         options = options
     );
-    test:assertEquals(oracledbClient.close(), (), "Client Error");
+    test:assertTrue(oracledbClient is error);
+    if oracledbClient is sql:ApplicationError {
+        test:assertTrue(oracledbClient.message().startsWith("Error in SQL connector configuration: Failed to initialize pool: " +
+        "IO Error: closing inbound before receiving peer's close_notify"));
+    } else {
+        test:assertFail("Error ApplicatonError expected");
+    }
+}
+
+// with all params and options with Erroneous SSL with correct port
+@test:Config {
+    groups:["connection"]
+}
+function testWithOptionsWithErroneousSSLCorrctPort() returns error? {
+     Options options = {
+            ssl: {
+                key: {
+                    path: clientStorePath,
+                    password: "password"
+                },
+                cert: {
+                    path: trustStorePath,
+                    password: "password"
+                }
+            }
+     };
+    Client|error oracledbClient = new(
+        host = HOST,
+        user = USER,
+        password = PASSWORD,
+        port = SSLPORT,
+        database = DATABASE,
+        options = options
+    );
+    test:assertTrue(oracledbClient is error);
+    if oracledbClient is sql:ApplicationError {
+        test:assertTrue(oracledbClient.message().startsWith("Error in SQL connector configuration: Failed to initialize pool: " +
+        "IO Error: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException"));
+    } else {
+        test:assertFail("Error ApplicatonError expected");
+    }
+}
+
+// with all params and options with SSL with correct port with incorrect password
+@test:Config {
+    groups:["connection"]
+}
+function testWithOptionsWithErroneousSSLCorrctPortWrongPW() returns error? {
+    string trustStorePathValid = check file:getAbsolutePath("./tests/resources/wallets/client-wallet/ballerina-server.p12");
+         Options options = {
+                ssl: {
+                    cert: {
+                        path: trustStorePathValid,
+                        password: "InvalidPassword"
+                    }
+                }
+         };
+    Client|error oracledbClient = new(
+        host = HOST,
+        user = USER,
+        password = PASSWORD,
+        port = SSLPORT,
+        database = DATABASE,
+        options = options
+    );
+    test:assertTrue(oracledbClient is error);
+    if oracledbClient is sql:ApplicationError {
+        test:assertTrue(oracledbClient.message().endsWith("The Network Adapter could not establish the connection Caused by " +
+        ":Unable to initialize ssl context."));
```

Comment:
Can we improve the message further. give exact error

Reply:
This error is originated in sql module and created an issue to track and update the error messages. https://github.com/ballerina-platform/ballerina-standard-library/issues/1775